### PR TITLE
✨ Add CLI tests

### PR DESCRIPTION
**Pull request by laurentsimon**
This adds tests for the CLI/main function.
We need additional tests for other errors (rekor, signatures, etc)

### CODE REVIEW

**Comment by laurentsimon** [comment]
> probably worth keeping the tests that match vtag/tag against a workflow_dispatch/no tag, but lgtm, thank you!

I have not removed them, or at least it was not my intention. Which did I remove?

**Comment by laurentsimon** [comment]
> > probably worth keeping the tests that match vtag/tag against a workflow_dispatch/no tag, but lgtm, thank you!
> 
> I have not removed them, or at least it was not my intention. Which did I remove?

my bad, some were commented out. Good catch

**Comment by laurentsimon** [comment]
> > > probably worth keeping the tests that match vtag/tag against a workflow_dispatch/no tag, but lgtm, thank you!
> > 
> > 
> > I have not removed them, or at least it was not my intention. Which did I remove?
> 
> my bad, some were commented out. Good catch

Fixed.

**Review by asraa — APPROVED** [review]

File: pkg/provenance.go [hunk 1]
```diff
@@ -426,59 +427,108 @@ func VerifyTag(env *dsselib.Envelope, expectedTag string) error {
...
-	semTag := strings.TrimPrefix(tag, "refs/tags/")
+	semTag := semver.Canonical(strings.TrimPrefix(tag, "refs/tags/"))
 	if !semver.IsValid(semTag) {
-		return fmt.Errorf("%s: %w", expectedTag, errorInvalidSemver)
+		return fmt.Errorf("%s: %w", expectedTag, ErrorInvalidSemver)
 	}
 
-	if semver.Compare(semTag, expectedTag) < 0 {
-		return errorMismatchVersionedTag
+	// fmt.Println("comparing", expectedTag, "with", semTag)
```
Comment (asraa) [inline pkg/provenance.go:451]: oops also leftover from debug!

Comment (laurentsimon) [reply pkg/provenance.go]: thanks.

**Review by asraa — APPROVED** [review]

**Pull request merged**